statedb/treeview.go

```diff
@@ -42,9 +44,11 @@ type TreeViewer interface {
 	// Iterate iterates over all leafs of this tree.  When callback returns true,
 	// the iteration is stopped and this function returns.
 	Iterate(callback func(key, value []byte) bool) error
-	// Dump exports all the tree leafs in a byte array.
-	Dump() ([]byte, error)
-	// Root returns the root of the tree, which cryptographically summarises the
+	// Dump exports all the tree leafs into a writer buffer.
```

Comment:
nit: this should probably not mention a buffer, as it takes a writer. how about:

> Dump exports all the tree leaves into w. The leaves can be read again via Import.

While you are here, you could also rename Dump to Export, which is the opposite of Import, and will be easier to understand.

Reply:
For consistency with the underlying package (Arbo) I think Dump is good. But thanks for the nit.